src/ape/utils/abi.py

```diff
@@ -5,7 +5,11 @@
 from eth_abi import decode, grammar
 from eth_utils import decode_hex, to_checksum_address
 from ethpm_types import HexBytes
-from ethpm_types.abi import ABIType, EventABI, EventABIType, MethodABI
+from ethpm_types.abi import ABIType, EventABI, MethodABI
+from pydantic import validator
+
+from ape.utils.basemodel import BaseModel
+from ape.utils.misc import cached_property
```

Comment:
This is a functools method, am I using the wrong wrapper?

Reply:
We export `@cached_property` from `ape.utils` for historical reasons, possibly because of Python 3.7 complications and to encourage its usage.